### DISCUSSION TITLE
Fork-related discussions

### DISCUSSION THREAD

**Discussion by p12tic** [input-leap/input-leap#1427] [General]
Here the maintainers will answer questions about the fork from Barrier. The situation was fairly well described in https://github.com/input-leap/input-leap/issues/1414 so we don't expect many questions, but we should have a place for discussions.

**Comment by win98se** [comment]
Will InputLeap be compatible with or upgradable from Barrier?

**Reply by shymega** [reply]
I believe it will be, yes, it would make it easier for Barrier users to migrate cleanly.

**Comment by the-spyke** [comment]
Good luck. I've used Synergy/Barrier every day and want it to improve both in stability (bugs) and in OS support (Wayland, Apple Silicon). I'd also like to contribute if you setup GH sponsorship.

**Reply by p12tic** [reply]
Unfortunately GitHub sponsorship is not an option due to how irresponsibly they handle taxes in the EU. If you want to support a specific developer, I have a Patreon for all the projects I work on: https://www.patreon.com/p12tic.

**Reply by the-spyke** [reply]
What about [OpenCollective](https://opencollective.com/)?

**Reply by darkdragon-001** [reply]
Many great projects live on OpenCollective, it is easy to use, transparent, they have an [EU based fiscal host](https://opencollective.com/europe), don't require a legal entity and make it easy to spend the money again with a virtual credit card.

**Reply by shymega** [reply]
The only thing with the EU host is that I live in the UK, and we have left the EU - so I'm not sure that'd work. But I'm with @p12tic on GitHub Sponsorship not being something I'd want to sign up for, primarily because of the tax handling.

**Reply by darkdragon-001** [reply]
OpenCollective works differently: You choose a fiscal host which will represent you as a legal entity. Often these fiscal hosts are non-profit and in turn don't have to pay taxes. You can then spend the money directly via OpenCollective (e.g. via virtual credit card or handing in invoices). What about some server space? Or a new work laptop? The collective can buy that for you! When you want to transfer some money to your private person, of course one needs to pay taxes. There are multiple ways of doing this but this is something you can still think about in the future when a big enough amount to justify the work has been collected. I think you should start collecting money first such that we come to that point earlier.

**Reply by shymega** [reply]
That's cool. I had considered GH Sponsors, but I'll create a OC account for myself. Cheers for the heads-up.

**Comment by jrbgit** [comment]
Found out about the fork while search for a way to do #824 - Looking forward to this release and ongoing dev. Also, I agree that getting away from the previous repos name is a good move professionally speaking. I had to google the word to find out what it meant, lol. cheers.

**Comment by darkdragon-001** [comment]
Why did you not use the _Github Fork_ button to fork the project?

This way, it would be discoverable via [_Insichts > Forks_](https://github.com/debauchee/barrier/network/members) and show the origin below the title for people to easily identify what this project is about.

**Reply by shymega** [reply]
That'd be up to @p12tic to answer, but I believe the main reason was for a fresh start.

**Reply by darkdragon-001** [reply]
You build on top of the existing code base though, right? Could be the same _fresh start_ when pressing the _fork_ button :wink: 
I highly appreciate your efforts!

**Reply by shymega** [reply]
Sure, but it's too late now. Maybe it should have been done, maybe not. I didn't create the repo :) Thank you!

**Reply by ssaadh** [reply]
Curious why it wasn't forked too. I don't mind or care. Just wondering why

**Reply by Rondom** [reply]
I think it might still be possible to add the fork-relation using a support-ticket to Github.

**Reply by p12tic** [reply]
There are two different meanings of the "fork" word. In github, the fork creates a repository that is intended to be as a staging ground for changes which will ultimately appear in the main upstream repository. When we say we forked Barrier, we mean that from that point, InputLeap and Barrier are completely separate. The code will not move between each other automatically. We don't intend to keep automatic compatibility (e.g. Barrier may decide to block connections to/from InputLeap). Thus keeping fork relation would only confuse users.

**Reply by shymega** [reply]
Agreed with @p12tic, I'm also not sure if Barrier will always remain compatible with Input Leap. Let's hope so, but things could always change.

**Reply by xpufx** [reply]
Barrier was forked from synergy and doesn't seem to have been adversely affected by the things you're citing as problems.  One really important thing I as a user would gain from this being a proper github fork would be the ability to see how many commits this project is ahead or behind barrier. 

When I look at a fork and it's significantly ahead of its upstream that's a sign that things are happening here now. Usually if a fork is significantly behind that's a personel fork for a PR or just to play with it. If there are a high number of differences in both ahead and behind that means the projects have diverged after the fork. 

It would be more open sourcey if you made the project discoverable with a fork and gave the users the chance to decide for themselves where the future of synergy lies. 

As it is you're having to do a lot of maintenance copying over github issues etc. If people knew (i totally found this "fork" by accident and even its README is still refering people back to barrier) there was a new kid in town they'd decide where to file issues themselves and you wouldn't have to do any unnecessary maintenance.

All in all if it's possible to mark this project as a fork as @Rondom says, please consider doing it. Or just create a new project, clean up all mention of barrier, and come out as a new project with no historical baggage.

**Reply by shymega** [reply]
Barrier was forked from Synergy, yes. But the problem was control over the Barrier repo and org, and that is why we branched away. We needed control for CI, and to have only one owner who wouldn't give any further ownership, was limiting. We're not going to mark this project as a GitHub fork, given Barrier (or, to play devil's advocate, Input Leap) could change the protocol to make a breaking change with vice versa. The README will be updated, yes. 

Unless you were a developer on the Barrier project and had the issues @p12tic and I encountered, I'm afraid this isn't something that will be fully understood.

**Reply by mitchcapper** [reply]
> We're not going to mark this project as a GitHub fork, given Barrier (or, to play devil's advocate, Input Leap) could change the protocol to make a breaking change with vice versa. The README will be updated, yes.

I think there are plenty of reasons to not be a GH fork (as it really treats it as a second class citizen) but there is nothing that would force Input Leap to take a breaking protocol change (or expectation for it to maintain protocol compatibility) just because it is a fork.  I strongly agree with the non-fork and think the readme notice about origins is certainly a better way to go.

**Comment by ssaadh** [comment]
Do maintainers have access to editing the ReadMe? Shouldn't there be a notice that development has moved here?
Also the ReadMe here still says barrier every where as the app name.

Finally, wouldn't it make sense to re-release the latest Barrier version but with Input Leap as the name? Then have that in the releases here, on Homebrew cask, and it makes a big text announcement on Barrier's ReadMe more seamless.

**Reply by p12tic** [reply]
Barrier still has an owner who thinks that there will be separate development in Barrier repository, so we can't just direct Barrier users to another repository. Though if InputLeap has more active development team this will happen anyway.

**Reply by blalond** [reply]
@p12tic I think he meant an edit to readme of input-leap would help clarify things. Currently it's the same as the old Barrier repo. https://github.com/input-leap/input-leap/blob/master/README.md

**Reply by shymega** [reply]
Yes, this is something I've got on a local branch.

**Reply by dJani97** [reply]
It took me a long time to find this fork. I had to browse through several issues on Barrier before I came across one reference that maintainers have moved to this repository. Since it has been a long time and Barrier is still not maintained, it would be really to add a warning / pinned issue / some readme text in the Barrier repository, directing users here.

**Reply by shymega** [reply]
@dJani97 Can't add a redirect link. As far as Barrier's concerned, it's a separate autononmous project.

**Reply by darkdragon-001** [reply]
@shymega looks more abandoned than autonomous to me...

**Reply by shymega** [reply]
@darkdragon-001 I didn't say it wasn't abandoned; it really is, but it's still separate to Input Leap. That's why we can't advertise Input Leap on Barrier's side. That's also another reason we can't accept issues of Barrier on this repo, as Barrier has had PRs merged that we don't have in our codebase.

**Reply by darkdragon-001** [reply]
@shymega if it is abandoned, then nobody should be against such a link in the `README.md` or a linked issue :stuck_out_tongue_winking_eye:

**Reply by shymega** [reply]
@darkdragon-001 Sorry, but as far as I'm concerned, we don't have write access to Barrier, the original owner still has control over it, and I don't want to stir flames by adding a link. That's the end of the matter, to be honest...

**Reply by darkdragon-001** [reply]
@shymega I thought the original maintainer doesn't respond any more, so he obviously doesn't care.

**Reply by shymega** [reply]
@darkdragon-001 He doesn't, but does that give us the right to advertise on their project? It doesn't. As far as I'm concerned now, I've addressed your point.

**Reply by ccoenen** [reply]
Inconvenient as it may be, I absolutely agree with @shymega.

**Comment by sheepdestroyer** [comment]
Just my 2cents about the "debauchee" matter mentionned in https://github.com/input-leap/input-leap/issues/1414

All other points make sense to me, but saying that this particular word is not "professional" and thus inappropriate is : 
- Strongly debatable (unless you live by the US/UK's prude entender of that term, as I take it to be, and argue that this should be the norm worldwide), but a lot more importantly 
- Not relevant. This is a community project, right? Why would professionalism be of concern then?

**Reply by shymega** [reply]
As someone who lives in the UK, I would tend to agree with @p12tic. Additionally, I put my OSS work on my CV, and having `debauchee` name doesn't look good. Even if it's a community project. The main overwhelming issue was control though.

**Reply by Unknown-Guy** [reply]
When I look at OSS projects popularity and open issue count is the main ones, so stating name as one of the big problems when you have glaring 660 (or 750 on original) open issues and something as fundamental as UTF8 support for this kind of project has lot of problems, kinda have to roll my eyes. But hey, making statements are not easy. The project is very useful and needs some love, so good luck getting things under control.

**Reply by shymega** [reply]
@Unknown-Guy I agree that UTF-8 is important, but the main issue is control. There was only one owner, and permissions were scarce. The bus factor was a huge decision in forking.

**Reply by luser** [reply]
You can debate this point all you want, but the maintainers are the ones doing the work here, so their opinion is what matters in practice. I think moving to a GitHub organization is a smart choice for long-term stability, since it means that new contributors/maintainers can easily be added without requiring a single repository owner to manage that.

Aside: I'm glad to see this project gaining traction! I've been a long-time Synergy user, probably for 15 years now, and I have been really disappointed by the choices that Symless has made over the years. I understand that maintaining software has a cost, and I even purchased a Synergy license ages ago when that was an option, but I don't feel like that has produced good results for Synergy as a piece of software.

**Reply by milesrout** [reply]
Interesting that you (@p12tic) prioritise an issue of "unprofessional" naming (when most people don't look twice at the name or closely inspect it for alternate meanings), when 2.5 years after collecting more than $2900 on BountySource for Wayland support in Barrier you still haven't come close to providing it. Are you going to give those people back their money? https://github.com/debauchee/barrier/issues/109#issuecomment-696715175

**Reply by p12tic** [reply]
@milesrout The issue with unprofessional naming is not about the users who indeed don't care, but about opinion of people who have authority to sign $x00000 consulting contracts. As a consultant I have to care about such details.

These contracts is also the reason why I haven't had enough time for implementing the feature. I should not have asked for donations at all and just did it for free at some point. The size of the donation is so low compared to my market rate as it's still effectively working for free, yet people now have expectations.

Note that the bounty is free to grab for anyone. I don't have any access to it. Also note that I didn't promise any deadline, but I understand that people are upset because by the spirit of the proposal the feature should not have taken so long to implement.

**Reply by ccoenen** [reply]
The bounty has not been paid out, so the term "collecting donations" might give people the wrong idea - especially when english is not their native language and/or translations might confuse collecting and cashing out (as is the case in german "spenden einnehmen"). The final sentence "give those people their money back" emphasizes this interpretation further.

At the time of writing, this bounty is still still with bountysource, and could be paid out to anyone working on it. Nobody has taken anything (well, except bountysource). Bountysource comes with its own advantages and disadvantages and is highly debatable, but this is another topic. The bounty idea was brought up by [a different person](https://github.com/debauchee/barrier/issues/109#issuecomment-652764762), and the bounty itself was posted by yet [another person](https://github.com/debauchee/barrier/issues/109#issuecomment-681989241). I don't see why @p12tic should need to explain himself?

**Reply by shymega** [reply]
Thanks for your comment @ccoenen, it's all good points. I'm sure @p12tic apperciates your support, as do I.

**Reply by shymega** [reply]
> @p12tic You asked for money to do something. If you set the rate too low that's your problem. Implement it or give the money back with interest.

He never took the money. It's bountysource that have it. With all due respect, I don't know how you can speak that way to someone you don't even know.

**Reply by p12tic** [reply]
@milesrout I appreciate your demand for justice. I agree with your complaints. The feature is in progress, multiple people have spent significant amount of time. Unfortunately unforeseen complications may arise thus I can't promise any dates about when the feature will be done.

**Reply by ccoenen** [reply]
(Links go to Bountysource's FAQ document, where these points may be explained in a little more detail.)

@milesrout I really think there's a misunderstanding here what a "bounty" means.
- First and foremost: anyone could work on this and claim the bounty (including you or me).
- Secondly: with the original project maintainer missing in action, it's very likely that the bounty would never even pay out (because [someone needs to confirm](https://github.com/bountysource/core/wiki/Frequently-Asked-Questions#do-i-have-to-be-affiliated-with-a-project-in-order-to-claim-a-bounty-on-an-issue) that this has been accomplished; most people would probably only confirm **after** this feature has been merged - unmerged code benefits very few people).
- Thirdly: [anyone who wants a refund can contact bountysource's support](https://github.com/bountysource/core/wiki/Frequently-Asked-Questions#can-i-receive-a-refund-for-a-bounty-i-posted). This has nothing to do with either the original project or this fork. Nor does this involve any of the people currently involved in the discussion. This is only between bountysource and the person pledging money to the issue.

**Reply by shymega** [reply]
@milesrout OK, so you seem to fundamentally misunderstand Bountysource. Povilas is *not* solely responsible for the funds collected. Please do not attack members of this project. Wayland is a complex beast, and there's been work on the XDG side of things with a new interface PR - flatpak/xdg-desktop-portal#714 by @whot, who has done an excellent job on this draft PR.

@ccoenen has explained the Bountysource side of things, but I will not tolerate your attitude towards members of this project. Povilas has done nothing wrong, he is NOT obliged to work on this project, and could easily walk away. So could I. This is the problem with open-source, people believe things are owed to them. This is simply not the case for a lot of projects. I have felt burned out before, and I had hoped Input Leap wouldn't be a repeat of my burning out. 

A promise isn't a formal contract, and as Povilas said, unforeseen complicates may arise.

Consider this a first warning, please do not attack members of this project, especially when it would appear you don't quite understand Bountysource. This is not worth attacking people over. It's Wayland, yes, it's a important feature, but no one is obliged to work on it.

@p12tic I'm sorry you had to see my response here, but I don't want you to be attacked for someone you're not legally responsible for, and I won't tolerate attacks against members of this project.

**Reply by p12tic** [reply]
@milesrout I have already agreed that offering to implement the feature was a mistake. What else do you want?

**Reply by G2G2G2G** [reply]
the repo name/guy's name is irrelevant, if they want to change it every week it wouldn't matter to me (would matter to search engines though :^] )
Since he's using it on resume etc it makes sense to change it, otherwise doesn't really matter either way.

UTF8 support definitely takes more work than uploading something in a different name

**Comment by spupuz** [comment]
sorry when there will be a release?

**Reply by shymega** [reply]
Not quite yet. I'm trying to sort out the copyright headers.

**Reply by dayeggpi** [reply]
just to get an idea, are we talking about days, weeks, month ? any way we can help (non dev) ? just found out about this and quite excited ! thanks for the work @shymega

**Reply by shymega** [reply]
@hycday If you look at the sources, you'll see copyright headers at the top of each file. This varies from file to file, so I can't do a simple sweeping change, and it has to be done incrementally. I'm finding that difficult to preserve each file's original contents. I don't think there's much non-devs can do. PRs are welcome, we have some specific requirements tho

**Reply by kohend** [reply]
Can you share the requirements? I'm a dev and would prefer not to use my own build...

**Reply by mitchcapper** [reply]
It looks like there is some attempt to potentially remove the old copyright where potentially all contributors approve doing so.   Would the easier solution not to just put a date range note at the top add a input leap copyright header and maintain the original ?

**Reply by shymega** [reply]
> It looks like there is some attempt to potentially remove the old copyright where potentially all contributors approve doing so. Would the easier solution not to just put a date range note at the top add a input leap copyright header and maintain the original ?

We've done this now, but there's more to be done.

**Reply by shymega** [reply]
> Can you share the requirements? I'm a dev and would prefer not to use my own build...

We've sorted the copyright headers now. Waiting for a few more PRs to be merged, and I'd like to get Qt 6 into v3.0.0 as well.

**Reply by i1u5** [reply]
Any ETA on this? Barrier is really starting to show its age with a lot of issues lately, would like to migrate (even if it's a pre-release), I appreciate your work though.

**Reply by shymega** [reply]
@i1u5 No update. I've really been trying to usher things along. A pre-release wouldn't be suitable at this stage.

**Reply by nickshouse** [reply]
What's the most fixed up version of Barrier that I can use in the meantime while this one is being worked on?

**Reply by Unknown-Guy** [reply]
@nickshouse I went back to Synergy, there is precompiled binaries here [DEAKSoftware/Synergy-Binaries](https://github.com/DEAKSoftware/Synergy-Binaries) works great with some minor issues I already had with Barrier. Not sure why I switched, but think I had some issue on MacOS.

One of the reasons for fork was said to be "operational issues maintaining", but from a users perspective more than a year and half later, there has been no releases, so 0 issues resolved compared to Barrier.

**Reply by shymega** [reply]
@nickshouse 

You are more than welcome to use Barrier v2.3.4, or at minimum - due to security patches - v2.3.3

**Reply by shymega** [reply]
@Unknown-Guy It's up to you what software you use. But please, don't trivialise our efforts. It's done entirely in our free time, we've made great strides compared to Barrier - so I'd ask for some understanding of our efforts.

We do not get paid for this.

**Reply by Unknown-Guy** [reply]
@shymega I am not trivializing your effort. But the end user's perspective is trivial - at bare minimum software with core features working.

**Reply by SpiralBrad** [reply]
> @\Unknown-Guy It's up to you what software you use. But please, don't trivialise our efforts.

Deservedly questioning the lack of public progress is not trivializing your efforts.

If you can make 'progress', you can make a release, yet here we are.

**Reply by shymega** [reply]
Sure. However, again, for volunteer projects, we don't do this for money, we do it because we choose to. However, I'm burning out rapidly. The release will come when we are ready. 

That much is all I can say.

**Reply by SpiralBrad** [reply]
> The release will come when we are ready.

...Will you ever be ready though?  It is absolutely a fair question at this point.

**Reply by the-spyke** [reply]
It would be nice if we could find a full time maintainer. Many (as me) use Barrier at work and I'm sure would gladly agree to pay a couple of bucks a month subscription to this person.

**Reply by shymega** [reply]
Who can say. I'm not a full-time maintainer, I'm job-hunting. I can't devote as much time as I used to for Input Leap/Barrier, and the release will come when it's ready. The best option for those who want a release faster is Synergy by Symless, who have the money to pay a full-time maintainer.

**Reply by Unknown-Guy** [reply]
> Who can say. I'm not a full-time maintainer, I'm job-hunting. I can't devote as much time as I used to for Input Leap/Barrier, and the release will come when it's ready. The best option for those who want a release faster is Synergy by Symless, who have the money to pay a full-time maintainer.

Thank you for setting clear expectations.

**Reply by shymega** [reply]
There is one other maintainer, who has more time for this project... I don't want to push people away, but equally, I need to make clear that this is a volunteer project, and compared to products like 'Mouse without Borders', or Symless's Synergy, we don't get that monetary compensation - which is necessary for living and motivation, too.

We might setup an OpenCollective fund soon as well. But for now, release will come when it's ready.

**Reply by the-spyke** [reply]
@shymega What if Barrier would become your job? We can try to do a poll here/reddit/yc/etc to get sense of how many people would like to subscribe.

**Reply by shymega** [reply]
@the-spyke There's a Wayland bounty, for example, was that the sort of subscription-type thing you were envisioning? But for maintainership of Input Leap itself? I'd like to do a poll soon on the future of the internals, to be honest.

**Reply by the-spyke** [reply]
@shymega I've contributed to that Wayland bounty :) My idea is that one-time payment is not sustainable. InputLeap needs constant work because of huge OS/DE support matrix and their continuous changes. So, I would advocate for at least something like Patreon where you can subscribe for $2-3 a month. I think we could get a couple of thousands of patrons to meet your salary demands. OpenCollective would be okay as well.

**Reply by shymega** [reply]
@the-spyke Yeah, I don't know what's happening with that bounty now that we have `libei`.

I can consider something like that, yes, but I'm more focusing on different areas of Input Leap currently. It might be a while before that branch is ready.

**Reply by shymega** [reply]
I have now been offered a cyber-security contract, haven't signed yet. I don't think donations towards Input Leap could sustain either me or @p12tic. There's too much possibility for fluctuations in donations. However, it would be useful to have OpenCollective. We're based in Europe (I'm the UK, so not EU now), Are there any good 'fiscal hosts' for that region?

**Reply by darkdragon-001** [reply]
There are
- https://opencollective.com/opensource
- http://opencollective.com/europe
- http://opencollective.com/uk
- https://opencollective.com/search?isHost=true

**Reply by shymega** [reply]
Thank you.

**Comment by resuna** [comment]
What's the plan for a release, or even maybe a final release of barrier to roll up the certificate fix?

**Reply by shymega** [reply]
Barrier is no longer maintained by anyone, not even me or anyone else in the IL team. The release will come when it's ready, for Input Leap.

**Reply by ansidev** [reply]
Could we have the preview build for testing?

**Reply by shymega** [reply]
@ansidev The preview build is non-existent. You can build from Git, which might be what you want.

**Reply by ansidev** [reply]
Do we have any plan for supporting brew.sh? I am seeing that MacPort is supported.

**Reply by shymega** [reply]
@ansidev My preference is that we, the maintainership team, do not manage every conceivable package manager. We will be managing things like Flatpak, and Snap, and I've registered namespaces for Chocolately, and Scoop.

I think there is a Homebrew package for Barrier, so we can adapt this. We do not manage MacPorts, however, and I need to balance the amount of upstream v. downstream efforts.

**Comment by ansidev** [comment]
One thing that I guess we can improve for input-leap is the documentation.
I am using barrier, and I struggle with troubleshooting any issue because of lacking documentation.

**Reply by shymega** [reply]
> One thing that I guess we can improve for input-leap is the documentation. I am using barrier, and I struggle with troubleshooting any issue because of lacking documentation.

You're more than welcome to make changes to the `wiki-prs` repository in the org. That's where the documentation belongs and via doxygen/rustc. Soon, we could merge the two together!

**Comment by mercster** [comment]
I promise I'm not trying to "pile on" here, just wondering what the best open source project to use at this point... it seems input-leap was forked from barrier because barrier was being unmaintained, but that not input-leap is also not moving at all?  To someone who hasn't been following whatever drama this is all about, can someone concisely state the differences here?

**Reply by huyz** [reply]
TLDR just install one of these https://github.com/DEAKSoftware/Synergy-Binaries

wish I had known about it before

**Reply by mercster** [reply]
Huh... weird.  I paid for Synergy 1 and was happy to do so... but then it became a pain and they were trying to push Synergy2 which looked horrible.  Then these open source things came out that looked almost identical in UI, I guess they are forks of a sort?  How did this situation get so odd?  😂  Did the Synergy guy lose control of the source?  Anyway, OK... I reckon I'll have a look at that.  Cheers.

**Reply by ccoenen** [reply]
Synergy was originally open source. At some later point, Nick Bolton (founder of Symless) came into the project and worked on it to some extent. Even later than that, Symless was founded, and binaries were sold. But the original project was GPL from the start. I made a longer post about it somewhere, but I can't find it right now.

**Reply by ccoenen** [reply]
here's a longer write-up: https://github.com/input-leap/input-leap/discussions/1633

**Reply by mercster** [reply]
Ahh, gotcha... that's very odd, cuz I have been using Linux and other open source tech since the early/mid 90s, you'd think I would have stumbled on the open source version first.  I didn't, I was looking for "software KVM" ages ago and stumbled across the Symless store page.  ¯\_(ツ)_/¯  Oh well, explains all these other variations floating around with very similar GUIs.  Thanks.

**Reply by mercster** [reply]
Ahh, very helpful. Cheers.

**Comment by mlabbe** [comment]
I don't mean to be rude or act entitled, but is this fork actively pursing its originally stated goals?  It was announced in Oct, with the last release being on October 12th.  As of this writing that's about half a year ago.  

I can see there have been a few pull requests merged since then.

Thanks - figured it was time to check in on status, as Input-Leap/Barrier/Synergy are programs that need to be installed on multiple computers per user, it would be good to know if this is an investment users should make.

**Reply by G2G2G2G** [reply]
Synergy re-open sourced their project and renamed it to deskflow https://github.com/deskflow/deskflow (you can see in commit history 10 years ago it is indeed synergy) 

Since barrier died and this fork basically failed, they probably felt bad.

**Reply by sithlord48** [reply]
This project is dead https://github.com/input-leap/input-leap/discussions/2199 . Use deskflow instead its active.
